pkg/addons/ensure.go

```diff
@@ -35,25 +35,26 @@ var (
 	// embeddedAddons is a list of addons that are embedded in the KubeOne
 	// binary. Those addons are skipped when applying a user-provided addon with the same name.
 	embeddedAddons = map[string]string{
-		resources.AddonCCMAws:             "",
-		resources.AddonCCMAzure:           "",
-		resources.AddonCCMDigitalOcean:    "",
-		resources.AddonCCMHetzner:         "",
-		resources.AddonCCMOpenStack:       "",
-		resources.AddonCCMPacket:          "",
-		resources.AddonCCMVsphere:         "",
-		resources.AddonCNICanal:           "",
-		resources.AddonCNICilium:          "",
-		resources.AddonCNIWeavenet:        "",
-		resources.AddonCSIAwsEBS:          "",
-		resources.AddonCSIAzureDisk:       "",
-		resources.AddonCSIAzureFile:       "",
-		resources.AddonCSIHetzner:         "",
-		resources.AddonCSIOpenStackCinder: "",
-		resources.AddonCSIVsphere:         "",
-		resources.AddonMachineController:  "",
-		resources.AddonMetricsServer:      "",
-		resources.AddonNodeLocalDNS:       "",
+		resources.AddonCCMAws:              "",
+		resources.AddonCCMAzure:            "",
+		resources.AddonCCMDigitalOcean:     "",
+		resources.AddonCCMHetzner:          "",
+		resources.AddonCCMOpenStack:        "",
+		resources.AddonCCMPacket:           "",
+		resources.AddonCCMVsphere:          "",
+		resources.AddonCNICanal:            "",
+		resources.AddonCNICilium:           "",
+		resources.AddonCNIWeavenet:         "",
+		resources.AddonCSIAwsEBS:           "",
+		resources.AddonCSIAzureDisk:        "",
+		resources.AddonCSIAzureFile:        "",
+		resources.AddonCSIHetzner:          "",
+		resources.AddonCSIOpenStackCinder:  "",
+		resources.AddonCSIVsphere:          "",
+		resources.AddonMachineController:   "",
+		resources.AddonMetricsServer:       "",
+		resources.AddonNodeLocalDNS:        "",
+		resources.AddonDefaultStorageClass: "",
```

Comment:
1. This map is used to list addons that are deployed by KubeOne automatically (e.g. CNIs, CCMs/CSIs, etc.). Addons listed here shouldn't appear in the configuration manifest unless the user wants to change some settings (however, no addon currently supports this).
2. `default-storage-class` is not the only embedded addon, we have a few others as well.